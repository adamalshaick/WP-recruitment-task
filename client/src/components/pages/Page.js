import React, { Component } from "react";
import fetchPlanetsPage from "../hoc/fetchPlanetsPage";
import PropTypes from "prop-types";
import Planet from "../planets/Planet";
import PlanetInfo from "../planets/PlanetInfo";
import {
  Table,
  Wrapper,
  TableElement,
  TableRow,
  TableName,
  TableHeader
} from "../common/styles/Table";
import { PlanetWrapper } from "../common/styles/Planet";

class Page extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showPlanetInfo: false
    };
  }

  onClick = () => {
    this.setState(prevState => ({
      showPlanetInfo: !prevState.showPlanetInfo
    }));
  };
  render() {
    return (
      <Table>
        <TableHeader>
          <TableName>Name</TableName>
          <TableElement>Diameter</TableElement>
          <TableElement>Climate</TableElement>
          <TableElement>Gravity</TableElement>
          <TableElement>Population</TableElement>
        </TableHeader>
        {this.props.planets.planets.map(planet => {
          return (
            <Planet
              planet={planet}
              onToggle={this.onClick}
              toogled={this.state.showPlanetInfo}
            />
          );
        })}
      </Table>
    );
  }
}

Page.propTypes = {
  pageNumber: PropTypes.number.isRequired
};

export default fetchPlanetsPage(Page);
