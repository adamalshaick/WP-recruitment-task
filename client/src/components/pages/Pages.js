import React from "react";
import Page from "./Page";
import PropTypes from "prop-types";

export const Pages = ({ planets }) => {
  const pages = [];
  for (let i = 1; i <= Math.ceil((planets.count + 1) / 10); i++) {
    pages.push(<Page key={i} pageNumber={i} />);
  }

  return <div>{pages}</div>;
};

Pages.propTypes = {
  planets: PropTypes.object.isRequired
};

export default Pages;
