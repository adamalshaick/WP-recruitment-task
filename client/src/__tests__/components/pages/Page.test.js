import React from "react";
import { Page } from "../../../components/pages/Page";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<Page />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
