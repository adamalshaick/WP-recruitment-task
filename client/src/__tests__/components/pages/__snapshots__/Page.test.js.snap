// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<styled.table>
  <tbody>
    <styled.tr>
      <styled.td>
        Name
      </styled.td>
      <styled.td>
        Diameter
      </styled.td>
      <styled.td>
        Climate
      </styled.td>
      <styled.td>
        Gravity
      </styled.td>
      <styled.td>
        Population
      </styled.td>
    </styled.tr>
    <Planet
      planet={
        Object {
          "name": "name",
        }
      }
    />
    <Planet
      planet={
        Object {
          "name": "name2",
        }
      }
    />
  </tbody>
</styled.table>
`;
